allolib-wasm/include/al_WebGPUCollision.hpp

```
@@ -0,0 +1,361 @@
+    }
+
+    void destroy() {
+        mShader.destroy();
+        mParamsBuffer.destroy();
+        mCreated = false;
+    }
+
+    ~SDFCollisionSystem() { destroy(); }
+
+    /// Add a sphere collider. Returns index or -1 if full.
+    int addSphere(Vec3f center, float radius, float restitution = 0.5f) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)